sdk/src/builder.rs

```diff
@@ -3282,4 +3474,47 @@ mod tests {
         let ingredient = active_manifest.ingredients().first().unwrap();
         assert_eq!(ingredient.title(), Some("C.jpg"));
     }
+
+    #[test]
+    fn test_builder_add_ingredient_from_reader() -> Result<()> {
```

Comment:
Why return a `Result` from this test? You're using `.unwrap()` which will panic and I don't see any `?` exits.